### PR TITLE
feat(csi-16): added proxy caching logic

### PR DESCRIPTION
**Pull request by geka-evk**

### CODE REVIEW

File: src/domain/parties/getPartiesByTypeAndID.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+ *
+ * @description sends request to applicable oracle based on type and sends results to a callback url
+ *
+ * @param {object} headers - incoming http request headers
+ * @param {object} params - uri parameters of the http request
+ * @param {string} method - http request method
+ * @param {object} query - uri query parameters of the http request
+ * @param {object} span
+ */
+const getPartiesByTypeAndID = async (headers, params, method, query, span = undefined, cache) => {
```
Comment (oderayi) [inline src/domain/parties/getPartiesByTypeAndID.js:51]: `cache` should be optional or placed before `span`.

File: src/domain/parties/getPartiesByTypeAndID.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+          filteredResponsePartyList = response.data.partyList.filter(party => party.partySubIdOrType == null) // Filter records that DON'T contain a partySubIdOrType
+          break
+        case FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTIES_SUB_ID_GET:
+          filteredResponsePartyList = response.data.partyList.filter(party => party.partySubIdOrType === partySubIdOrType) // Filter records that match partySubIdOrType
+          break
+        default:
+          filteredResponsePartyList = response // Fallback to providing the standard list
+      }
+
+      if (filteredResponsePartyList == null || !(Array.isArray(filteredResponsePartyList) && filteredResponsePartyList.length > 0)) {
```
Comment (oderayi) [inline src/domain/parties/getPartiesByTypeAndID.js:122]: The last part of this condition is not clear. `filteredResponsePartyList.length > 0` will never be true if `filteredResponsePartyList` is not an array, i.e. the condition checked before `!(Array.isArray(filteredResponsePartyList)`.

File: src/domain/parties/getPartiesByTypeAndID.js (hunk 1 above)
Comment (geka-evk) [reply src/domain/parties/getPartiesByTypeAndID.js:51]: it's not my impl., it's legacy code:
https://github.com/mojaloop/account-lookup-service/blob/feat/fx-impl/src/domain/parties/parties.js#L55

File: src/domain/parties/getPartiesByTypeAndID.js (hunk 2 above)
Comment (geka-evk) [reply src/domain/parties/getPartiesByTypeAndID.js:122]: it's not my impl., it's a legacy code:
https://github.com/mojaloop/account-lookup-service/blob/feat/fx-impl/src/domain/parties/parties.js#L113

**Pull request merged**